### PR TITLE
bpo-37893: raise ValueError in pow if exponent is negative and modulus is +-1

### PR DESCRIPTION
**Pull request by isidentical**
Check modulus and the exponent and if there is a case like exponent is negative and abs(modulus) is 1, raise a ValueError

https://bugs.python.org/issue37893

### CODE REVIEW

File: Objects/longobject.c [hunk 1]
```diff
@@ -4334,6 +4333,14 @@ long_pow(PyObject *v, PyObject *w, PyObject *x)
                 goto Error;
         }
 
+        /* if exponent < 0 and abs(modulus) == 1:
+               raise ValueError */
+        if ((Py_SIZE(b) < 0) && (c->ob_digit[0] == 1)) {
```
Comment (tim-one) [inline Objects/longobject.c]: Looks like this is only testing whether the least significant digit of the modulus is 1, not whether the entire modulus is 1.  Like the test below this, I expect this needs to also test for `Py_SIZE(c) == 1`.

**Review by tim-one — CHANGES_REQUESTED** [review]
Thanks!  Added a comment about what appears to be a logic error.

**Comment by mdickinson** [comment]
I've already commented on the issue, but for the record I'm (quite strongly) opposed to changing this. It's perfectly correct as it is.

**Comment by isidentical** [comment]
I have made the requested changes; please review again

**Review by tim-one — COMMENTED** [review]
Thanks again!  However, on the bpo report, Mark is opposed to this change, so I expect we'll have to abandon this.  Sorry for the hassle!

**Comment by tim-one** [comment]
I'm closing this, since Mark gave a good case for leaving it alone.  Thank you for the effort!

**Pull request closed without merging**